platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -306,8 +306,23 @@ public List<ConsensusRound> addEvent(@NonNull final EventImpl event) {
             final List<ConsensusRound> rounds = new ArrayList<>();
             // set its round to undefined so that it gets calculated
             event.setRoundCreated(ConsensusConstants.ROUND_UNDEFINED);
-            checkInitJudges(event);
-            ConsensusRound consensusRound = calculateAndVote(event);
+            ConsensusRound consensusRound;
+
+            final boolean lastJudgeFound = checkInitJudges(event);
+
+            if (!noInitJudgesMissing()) {
+                // we should not do any calculations or voting until we have found all the init judges
+                return rounds;
```

Comment:
It would be clearer to return `Collections.emptyList()` instead of `rounds`.  I was confused at first that rounds might not be empty.

Reply:
makes sense, done. I also moved the declaration of rounds further down